OAT.Tests/CaptureTests.cs

```diff
@@ -0,0 +1,835 @@
+
+            ltCaptureRule = new Rule(RuleName)
+            {
+                Clauses = new List<Clause>()
+                {
+                    new Clause(OPERATION.LT)
+                    {
+                        Data = new List<string>()
+                        {
+                            "5"
+                        },
+                        Capture = true,
+                        Invert = true
+                    }
+                }
+            };
+
+            Assert.IsTrue(analyzer.GetCapture(ltCaptureRule, 7, null).Result?.Captures.Any(x => x is TypedClauseCapture<int> y && y.Result == 7) is true);
+        }
+
+        [TestMethod]
+        public void TestWasModifiedCapture()
+        {
+            var RuleName = "WasModified Capture";
+            var analyzer = new Analyzer();
+
+            var wasModifiedRule = new Rule(RuleName)
+            {
+                Clauses = new List<Clause>()
+                {
+                    new Clause(OPERATION.WAS_MODIFIED)
+                    {
+                        Capture = true
+                    }
+                }
+            };
+
+            var testString = "The Secret is Magic";
+            var testString2 = "The Secret is Science";
+            var res = analyzer.GetCapture(wasModifiedRule, testString, testString2);
+            Assert.IsTrue(res.Result?.Captures.Any(x => x is TypedClauseCapture<ComparisonResult> y && !y.Result.AreEqual) is true);
+
+            wasModifiedRule = new Rule(RuleName)
+            {
+                Clauses = new List<Clause>()
+                {
+                    new Clause(OPERATION.WAS_MODIFIED)
+                    {
+                        Capture = true,
+                        Invert = true
+                    }
+                }
+            };
+            res = analyzer.GetCapture(wasModifiedRule, testString, testString);
+            Assert.IsTrue(res.Result?.Captures.Any(x => x is TypedClauseCapture<ComparisonResult> y && y.Result.AreEqual) is true);
+        }
+
+        [TestMethod]
+        public void TestRegexCapture()
+        {
+            var RuleName = "Regex Capture";
+            var analyzer = new Analyzer();
+            var regexCapture = new Rule(RuleName)
+            {
+                Clauses = new List<Clause>()
+                {
+                    new Clause(OPERATION.REGEX)
+                    {
+                        Data = new List<string>()
+                        {
+                            "Secret.*([Mm]agic).*"
+                        },
+                        Capture = true
+                    }
+                }
+            };
+
+            var testString = "The Secret is Magic";
+            var res = analyzer.GetCapture(regexCapture, testString, null);
+            Assert.IsTrue(res.Result?.Captures.Any(x => x is TypedClauseCapture<Match> y && y.Result.Groups[1].Value == "Magic") is true);
```

Comment:
I'm sure this is correct, but what does the `is true` at the end of this line do?

Reply:
The result of that evaluation is truthy (bool?) But assert doesn't work on truthy values so you need to check if it's actually a bool, so I check is true - both not null and not false.